project/Sheng08/README.md

```diff
@@ -0,0 +1,304 @@
+efficient algorithms with an interactive visual interface. This combination
+not only simplifies OBB extraction but also enhances user comprehension
+through vivid visual representations.
+
+- 📂 **GitHub Repository**: [https://github.com/Sheng08/OrientMesh-Toolkit](https://github.com/Sheng08/OrientMesh-Toolkit)
+- 💻 **Implemented In**:
+  - **Core Library**: C++
+  - **Bindings & Interface(GUI)**: Python
+
+## Problem to Solve
+
+Triangular meshes, particularly when it comes to intricate surface detailing,
+can be both complex and computationally intensive. Operations on these meshes
+often demand significant resources. Having a precise oriented bounding box aids
+in simplifying many computational geometry problems related to surface meshes,
+like collision detection, mesh simplification, and many more. However,
+generating an optimal OBB for surface meshes is a non-trivial task. OrientMesh
+aims to offer a reliable, optimized, and easily-integrable solution for this
+challenge.
+
+<p align="center">
+    <img src="./doc/pic/OOB_mesh_sample1.png" alt="" height="250"style="margin-right: 15px;"/>
+    <img src="./doc/pic/OOB_mesh_sample2.png" alt="" height="250"/>
+</p>
+
+## Prospective Users
+
+_**OrientMesh Toolkit**_ targets game developers, computer science students,
+researchers, and professionals working with 3D models, especially in areas
+requiring accurate bounding boxes like game development, simulation, and
+augmented reality.
+
+## System Architecture
+
+_**OrientMesh Toolkit**_ utilizes C++ for its core computational tasks,
+ensuring performance optimization. Moreover, Python is employed for its
+bindings, making it seamlessly usable for a wider audience and also aiding
+in GUI development and design.
+
+**Components**:
+
+- **Triangular Mesh Data Structures (C++)**: This component serves as the
+  foundation, hosting the mesh structures essential for subsequent operations.
+- **Algorithm for Optimal OBB Extraction (C++)**: A dedicated component that
+  undertakes the crucial task of extracting the Oriented Bounding Boxes from
+  the meshes.
+- **API (C++ and Python)**: Interface that facilitates seamless communication
+  between different modules and allows extensibility for future needs.
+- **Python Bindings Using `pybind11`**: This bridges the gap between C++
+  computations and Python's versatility, opening up the toolkit for a wider
+  user base.
+- **Interactive GUI (Python `PyQt`)**: A visual platform enabling users to
+  visualize the mesh and extracted OBB interactively.
+
+**System Flow Description**:
+
+In the OrientMesh workflow, everything begins with the user providing 3D object
+data in the form of a `.obj` file, which encapsulates information about vertices,
+edges, and faces. Initially, this data is ingested and stored by the
+`MeshDataHandler` class into specialized C++ data structures. Throughout this
+process, we leverage both the `glm` library (OpenGL Mathematics[^3]) for
```

Comment:
When the implementation in the libraries is wrong or inconsistent (it happens often), what do you plan to do?  Please write a sentence or two to describe your plan.  In 8 weeks you may not get that deep.

Reply:
You are talking about how to make sure things are correct.  I was asking what do you do when your dependencies are wrong.  It's OK that you leave it an open question.  Just keep it in mind that everything can go wrong, including a compiler.